client/components/date-range/test/index.js

```diff
@@ -526,229 +459,195 @@ describe( 'DateRange', () => {
+			await userEvent.keyboard( '04/01/2018' );
+			await userEvent.click( screen.getByLabelText( 'To' ) );
+			await userEvent.keyboard( '04/29/2018' );
+			await userEvent.click( screen.getByLabelText( 'Apply' ) );
 
-			// Commit/apply those dates
-			wrapper.instance().commitDates();
+			const newRangeText = screen
+				.getByLabelText( 'Select date range' )
+				.querySelector( 'span' ).innerHTML;
 
-			expect( callback ).toHaveBeenCalledTimes( 1 );
-			expect( dateToLocaleString( callback.mock.calls[ 0 ][ 0 ] ) ).toEqual(
-				dateToLocaleString( newStartDate )
-			);
-			expect( dateToLocaleString( callback.mock.calls[ 0 ][ 1 ] ) ).toEqual(
-				dateToLocaleString( newEndDate )
-			);
+			expect( originalRangeText ).toEqual( 'MM/DD/YYYY - MM/DD/YYYY' );
+			expect( newRangeText ).toEqual( '04/01/2018 - 04/29/2018' );
 		} );
-	} );
 
-	describe( 'Actions and Information UI', () => {
-		test( 'should only persist date selection when user clicks "Apply" button', () => {
-			const wrapper = shallow( <DateRange translate={ translate } moment={ moment } /> );
-			const originalStartDate = wrapper.state().startDate;
-			const originalEndDate = wrapper.state().endDate;
-
-			// Get child components
-			const dateRangeHeader = wrapper.find( DateRangeHeader );
-			const datePicker = wrapper.find( DatePicker );
-
-			const newStartDate = moment( '2018-04-01' );
-			const newEndDate = moment( '2018-04-29' );
-
-			// Select dates using API
-			datePicker.props().onSelectDay( newStartDate );
-			datePicker.props().onSelectDay( newEndDate );
-
-			// Force update
-			wrapper.update();
-
-			// Calls whichever method DateRange passes into DateRangeHeader component
-			dateRangeHeader.props().onCancelClick();
-
-			// Should still be the original dates...
-			expect( wrapper.state() ).toEqual(
-				expect.objectContaining( {
-					startDate: originalStartDate,
-					endDate: originalEndDate,
-				} )
-			);
+		test( 'should discard date selection when user clicks the "Cancel" button', async () => {
+			render( <DateRange translate={ translate } moment={ moment } /> );
 
-			// Select dates using API
-			datePicker.props().onSelectDay( newStartDate );
-			datePicker.props().onSelectDay( newEndDate );
+			const originalRangeText = screen
+				.getByLabelText( 'Select date range' )
+				.querySelector( 'span' ).innerHTML;
 
-			// Force update
-			wrapper.update();
+			await userEvent.click( screen.getByLabelText( 'Select date range' ) );
+			await userEvent.click( screen.getByLabelText( 'From' ) );
+			await userEvent.keyboard( '04/01/2018' );
+			await userEvent.click( screen.getByLabelText( 'To' ) );
+			await userEvent.keyboard( '04/29/2018' );
+			await userEvent.click( screen.getByLabelText( 'Cancel' ) );
 
-			// Calls whichever method DateRange passes into DateRangeHeader component
-			dateRangeHeader.props().onApplyClick();
+			const newRangeText = screen
+				.getByLabelText( 'Select date range' )
+				.querySelector( 'span' ).innerHTML;
```

Comment:
Instead of using `.innerHTML` for grabbing a certain text, and using that in a `toEqual()` below, could we use `.toHaveTextContent()` instead?

Reply:
Sure, I'll try that!

Reply:
Done in https://github.com/Automattic/wp-calypso/pull/63879/commits/60dcbb0da6b4076cf54a47a97d2670bbd21a8281. Thanks for the RTL API tips! 🙇🏻